site/it/community/contribute/code.md

```diff
@@ -0,0 +1,136 @@
+
+Per tutti i dettagli particolari sulle procedure di test in ciascun progetto TensorFlow, vedere i file `README.md` e `CONTRIBUTING.md` nel rispettivo repository su GitHub.
+
+Di particolare importanza è fare *testing adeguato*:
+
+*   E stata testata *ogni funzione pubblica di ogni classe*? 
+*   Sono stati testati *insiemi ragionevoli di parametri*, di valori, tipi e in ragionevoli combinazioni? 
+*   I test sono in grado di validare che *il codice è corretto*, e che esso faccia *ciò che la documentazione dice* che esso faccia?
+*   Se la modifica è la correzione di un difetto, è incluso un *test di non regressione*?
+*   Il test *supera il build in continuous integration*?
+*   Il test *copre tutte le linee del codice?* se no, le eccezioni sono ragionevoli ed esplicite?
+
+Se individuate dei problemi, cortesemente, considerate la possibilità di supportare un contributore a capire questi problemi ed a risolverli. 
+
+
+## Miglioramento dei messaggi di errore o dei log
+
+Accogliamo contributi che migliorano la messaggistica di errore e la tracciatura. 
+
+
+## Flusso dei contributi
+
+Tutti i contributi al codice—correzioni, nuovi sviluppi, miglioramenti ai test—seguono un flusso centrato su GitHub. Per partecipare allo sviluppo in TensorFlow, aprite un account GitHub. Quindi:
+
+1.  Fate un Fork del repository su cui avete intenzione di lavorare.
+    Andate sulla pagina del repository di progetto e usate il pulsante *Fork*. Ciò creerà una copia del
+    repository, con il vostro nome utente. (Per maggiori dettagli su come fare il fork di un repository vedere
+    [questa guida](https://help.github.com/articles/fork-a-repo/).)
+
+2.  Clonate il repository sul vostro sistema locale.
+
+    `$ git clone git@github.com:your-user-name/project-name.git`
+
+3.  Definite un nuovo branch per isolare il vostro lavoro.
+
+    `$ git checkout -b new-branch-name`
+
+4.  Lavorate sul vostro nuovo codice. Scrivere ed eseguite i test.
+
+5.  Fate il commit delle vostre modifiche.
+
+    `$ git add -A`
+
+    `$ git commit -m "commit message here"`
+
+6.  Fate un push delle vostre modifiche sul vostro repository GitHub.
+
+    `$ git push origin branch-name`
+
+7.  Aprite una *Pull Request* (PR). Andando su repository originale del progetto su GitHub. Ci sarà un'evidenza a proposito del branch di cui avete fatto il push di recente, che vi chiederà se avete l'intenzione di aprire una pull request. Scegliete, *confronto tra repository* (*compare across repositories* n.d.t), e sottoponete la PR. Ciò produrrà l'invio di una email ai committer. Per dare maggiore visibilità potete considerare la possibilità di inviare anche un'email alla mailing list. (Per maggiori dettagli, vedere la [guida di GitHub sulle PR](https://help.github.com/articles/creating-a-pull-request-from-a-fork). 
```

Comment:
`una segnalazione in evidenza relativa al branch`

Reply:
Looking at the original English I prefer "Ci sarà un messaggio che evidenzia il branch". In my opinion it sounds better...

Reply:
It Is also ok